claripy/simplifications.py

```diff
@@ -280,6 +295,13 @@ def ne_simplifier(a, b):
             # elif b._claripy.is_true(b.args[1] == a) and b._claripy.is_true(b.args[2] == a):
             #	  return b._claripy.false
 
+        # 1 ^ expr != 0     ->   expr == 0
+        if a.op == '__xor__' and b.op == "BVV" and b.args[0] == 0 and len(a.args) == 2:
+            if a.args[1].op == "BVV" and a.args[1].args[0] == 1 and a.args[0].size() == 1:
+                return a.args[0] == 0
```

Comment:
Delete `and a.args[0].size() == 1` and replace `return a.args[0] == 0` with `return a.args[0] != 1` I think.

Unless you intend for this to only work for BVs of length 1-bit (functionally booleans). The logic is correct if that is desired, ping @ltfish for if we want to include this simplifier or if it is too specific or not.

Reply:
Good catch. Addressed.